### ISSUE TITLE
possible feature request - background subtraction of a 2D background image

### ISSUE THREAD

**Issue by mperrin** [STScI-JWST/wss_tools#8]
Also mentioned by Knight, but it's TBD if this is sufficiently a concern to implement.

Right now the background subtraction just works to subtract a constant background level, and that is expected to be sufficient for WFS analyses purposes. As a contingency capability, if there is more spatially variable background than we expect, it might be desired to have some capability to subtract a 2D background image. I.e. load an image A, then select some other image B (via a file selection dialog?) and subtract image B from image A.

**Comment by pllim** [comment]
Okay, I'll just leave this be until I hear more. If we go forward, it would be a *new* plugin just because the GUI would be very different from existing `BackgroundSub` (maybe name it `BgSub2D` or something). Extra details would be nice:

* What if selected image is too big/small?
* Do we need to check if it is the *right* kind of image (e.g., using specific header keywords)?
* Do you also want to propagate `ERR` and `DQ`?
* Do you want to set any special keywords in the subtracted image indicating that background has been subtracted (like what happens in a pipeline)? This is in addition to existing logging mechanism.

If you decide it's a no-go, then feel free to close.

**Comment by mperrin** [comment]
yes agreed that was not a fully detailed feature request! Just capturing this as a note in real time during the WFSCOWG meeting ongoing right now here in the board room!
